feat(standards): add ariaRoles standard

straker opened the pull request:
This is the first part of the standards role table. Since there are >100 roles we need to do, I think splitting it out into smaller part will make the review much easier. These are the first 25 roles. 

I haven't made any significant changes to the table other than those discussed in the proposal:

* flattened the `attributes` object to just be `allowedAttrs` and `requiredAttrs`
* flattened the `owned` object to just be an array
* removed the `allowedElements` property as that will be controlled by the [`html-elms` standards object](https://github.com/dequelabs/axe-core/pull/2325)

As a note, `aria-errormessage` is a global attribute in the 1.1 spec so isn't listed in any `allowedAttrs` for that reason.

Lastly, to help with the role table being so large, I think we should split the dpub roles into their own object (`dpub-roles`) and combine them with the aria-roles object into the `standards` object. They don't get used a lot and just take up room in the aria table (they're also a different spec).

Part of issues: #2108, #2303, and #2243

## Reviewer checks

**Required fields, to be filled out by PR reviewer(s)**

- [x] Follows the commit message policy, appropriate for next version
- [x] Code is reviewed for security

WilcoFiers commented on lib/standards/aria-roles.js:
```diff
@@ -0,0 +1,221 @@
+// Source: https://www.w3.org/TR/wai-aria-1.1/#roles
+
+/* easiest way to see allowed roles is to filter out the global ones
+   from the list of inherited states and properties. The dpub spec
+   does not have the global list so you'll need to copy over from
+   the wai-aria one:
+
+  const globalAttrs = Array.from(
+    document.querySelectorAll('#global_states li')
+  ).map(li => li.textContent.replace(/\s*\(.*\)/, ''));
+
+  const globalRoleAttrs = Array.from(
+    document.querySelectorAll('.role-inherited li')
+  ).filter(li => globalAttrs.includes(li.textContent.replace(/\s*\(.*\)/, '')))
+
+  globalRoleAttrs.forEach(li => li.style.display = 'none');
```
What's the point of this comment?

straker replied:
Trying to filter out the list of global attributes in the roles supported and inherited lists is difficult when all you want to see is the non-global ones for this object. Since I've had to look at spec a lot and manually filter them out, I wanted to leave a helpful comment for myself and future collaborators on how to see just the non-global attributes.

WilcoFiers commented on lib/standards/aria-roles.js:
```diff
@@ -0,0 +1,221 @@
+	alertdialog: {
+		type: 'widget',
+		allowedAttrs: ['aria-expanded', 'aria-modal'],
+		nameFrom: ['author']
```
Since we're changing the structure, can we get rid of this, and replace it with a `nameFromContent?: boolean`? (without the "S", ARIA calls it "name from content")

Everything is name from author, there's no reason to put that everywhere.

WilcoFiers commented on lib/standards/aria-roles.js:
```diff
@@ -0,0 +1,221 @@
+		type: 'structure',
+		allowedAttrs: ['aria-posinset', 'aria-setsize', 'aria-expanded'],
+		nameFrom: ['author'],
+		implicit: ['article']
```
We shouldn't need this anymore now that we have `implicitHtmlRole`. Looking through the code base, I think we need to update `aria.implicitNodes()`. It's a little hard to tell, but from the look of it that's the only place we're directly using this prop.

WilcoFiers commented on lib/standards/aria-roles.js:
```diff
@@ -0,0 +1,221 @@
+	},
+	combobox: {
+		type: 'composite',
+		owned: ['listbox', 'tree', 'grid', 'dialog', 'textbox'],
```
How about we do this. That's what ARIA calls them.
```suggestion
		requiredOwned: ['listbox', 'tree', 'grid', 'dialog', 'textbox'],
```

WilcoFiers commented on lib/standards/aria-roles.js:
```diff
@@ -0,0 +1,221 @@
+	},
+	columnheader: {
+		type: 'structure',
+		context: ['row'],
```
I think we should use ARIA terminology here.

```suggestion
		requiredContext: ['row'],
```

straker commented:
Now that the structure is finalized I added all the roles to the table, generating the list automatically using a custom script on the 1.0 spec page.

I then created a script to diff the generated file against our current lookup table and modified the output as needed to match (putting comments where and why changes were made).

The final output of the diff contains things I believe we are OK with but am putting here for reference. They are mostly additional aria allowed attrs that the current lookup table did not have but the spec allows.

```js
====== combobox.allowedAttrs ======
{ AssertionError [ERR_ASSERTION]: combobox.allowedAttrs does not match
    at file:///Users/stevenlambert/deque/axe-core/diff.mjs:55:14
    at ModuleJob.run (internal/modules/esm/module_job.js:95:12)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual:
   [ 'aria-activedescendant',
     'aria-autocomplete',
     'aria-controls',
     'aria-orientation',
     'aria-readonly',
     'aria-required' ],
  expected:
   [ 'aria-activedescendant',
     'aria-autocomplete',
     'aria-orientation',
     'aria-required' ],
  operator: 'deepEqual' }


====== menuitemcheckbox.allowedAttrs ======
{ AssertionError [ERR_ASSERTION]: menuitemcheckbox.allowedAttrs does not match
    at file:///Users/stevenlambert/deque/axe-core/diff.mjs:55:14
    at ModuleJob.run (internal/modules/esm/module_job.js:95:12)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual:
   [ 'aria-checked',
     'aria-posinset',
     'aria-readonly',
     'aria-setsize' ],
  expected: [ 'aria-checked', 'aria-posinset', 'aria-setsize' ],
  operator: 'deepEqual' }


====== menuitemradio.allowedAttrs ======
{ AssertionError [ERR_ASSERTION]: menuitemradio.allowedAttrs does not match
    at file:///Users/stevenlambert/deque/axe-core/diff.mjs:55:14
    at ModuleJob.run (internal/modules/esm/module_job.js:95:12)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual:
   [ 'aria-checked',
     'aria-posinset',
     'aria-readonly',
     'aria-setsize' ],
  expected:
   [ 'aria-checked',
     'aria-posinset',
     'aria-selected',
     'aria-setsize' ],
  operator: 'deepEqual' }


====== radio.allowedAttrs ======
{ AssertionError [ERR_ASSERTION]: radio.allowedAttrs does not match
    at file:///Users/stevenlambert/deque/axe-core/diff.mjs:55:14
    at ModuleJob.run (internal/modules/esm/module_job.js:95:12)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual:
   [ 'aria-checked',
     'aria-posinset',
     'aria-required',
     'aria-setsize' ],
  expected:
   [ 'aria-checked',
     'aria-posinset',
     'aria-required',
     'aria-selected',
     'aria-setsize' ],
  operator: 'deepEqual' }


====== scrollbar.allowedAttrs ======
{ AssertionError [ERR_ASSERTION]: scrollbar.allowedAttrs does not match
    at file:///Users/stevenlambert/deque/axe-core/diff.mjs:55:14
    at ModuleJob.run (internal/modules/esm/module_job.js:95:12)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual:
   [ 'aria-controls',
     'aria-orientation',
     'aria-valuemax',
     'aria-valuemin',
     'aria-valuenow',
     'aria-valuetext' ],
  expected:
   [ 'aria-orientation',
     'aria-valuemax',
     'aria-valuemin',
     'aria-valuetext' ],
  operator: 'deepEqual' }


====== separator.allowedAttrs ======
{ AssertionError [ERR_ASSERTION]: separator.allowedAttrs does not match
    at file:///Users/stevenlambert/deque/axe-core/diff.mjs:55:14
    at ModuleJob.run (internal/modules/esm/module_job.js:95:12)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual:
   [ 'aria-orientation',
     'aria-valuemax',
     'aria-valuemin',
     'aria-valuenow',
     'aria-valuetext' ],
  expected:
   [ 'aria-expanded',
     'aria-orientation',
     'aria-valuemax',
     'aria-valuemin',
     'aria-valuenow',
     'aria-valuetext' ],
  operator: 'deepEqual' }


====== slider.allowedAttrs ======
{ AssertionError [ERR_ASSERTION]: slider.allowedAttrs does not match
    at file:///Users/stevenlambert/deque/axe-core/diff.mjs:55:14
    at ModuleJob.run (internal/modules/esm/module_job.js:95:12)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual:
   [ 'aria-orientation',
     'aria-readonly',
     'aria-valuemax',
     'aria-valuemin',
     'aria-valuenow',
     'aria-valuetext' ],
  expected:
   [ 'aria-orientation',
     'aria-readonly',
     'aria-valuemax',
     'aria-valuemin',
     'aria-valuetext' ],
  operator: 'deepEqual' }


====== spinbutton.allowedAttrs ======
{ AssertionError [ERR_ASSERTION]: spinbutton.allowedAttrs does not match
    at file:///Users/stevenlambert/deque/axe-core/diff.mjs:55:14
    at ModuleJob.run (internal/modules/esm/module_job.js:95:12)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual:
   [ 'aria-activedescendant',
     'aria-readonly',
     'aria-required',
     'aria-valuemax',
     'aria-valuemin',
     'aria-valuenow',
     'aria-valuetext' ],
  expected:
   [ 'aria-readonly',
     'aria-required',
     'aria-valuemax',
     'aria-valuemin',
     'aria-valuetext' ],
  operator: 'deepEqual' }


====== switch.allowedAttrs ======
{ AssertionError [ERR_ASSERTION]: switch.allowedAttrs does not match
    at file:///Users/stevenlambert/deque/axe-core/diff.mjs:55:14
    at ModuleJob.run (internal/modules/esm/module_job.js:95:12)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual: [ 'aria-readonly' ],
  expected: [],
  operator: 'deepEqual' }


====== table.allowedAttrs ======
{ AssertionError [ERR_ASSERTION]: table.allowedAttrs does not match
    at file:///Users/stevenlambert/deque/axe-core/diff.mjs:55:14
    at ModuleJob.run (internal/modules/esm/module_job.js:95:12)
  generatedMessage: false,
  name: 'AssertionError [ERR_ASSERTION]',
  code: 'ERR_ASSERTION',
  actual: [ 'aria-colcount', 'aria-expanded', 'aria-rowcount' ],
  expected: [ 'aria-colcount', 'aria-rowcount' ],
  operator: 'deepEqual' }
```
